app/models/Appeal.scala

```diff
@@ -19,9 +19,21 @@ package models
 import models.AppealStatus.AppealStatus
 import models.AppealType.AppealType
 
-case class Appeal
-(
-  id: String,
-  status: AppealStatus,
-  `type`: AppealType
-)
+case class Appeal(
+                   id: String,
+                   status: AppealStatus,
+                   `type`: AppealType
+                 )
+
+object Appeal {
+
+  def highestAppealFromDecision(decision: Option[Decision]): Option[Appeal] = {
+    val appeals = decision.map(_.appeal).getOrElse(Seq.empty)
+
+    if (appeals.nonEmpty) {
+      Some(appeals.sortWith((a, b) => a.`type`.id > b.`type`.id).head)
+    } else {
+      None
+    }
```

Comment:
I think this could be implemented with `appeals.maxBy(_.``type``.id)`

`maxByOption` would allow you to do this even more easily, but unfortunately that was only introduced in Scala 2.13

For reference:
```scala
for {
  decision <- decisionOpt
  highestAppeal <- decision.appeals.maxByOption(_.`type`.id)
} yield highestAppeal
```

Reply:
great comment I will implement this, thanks